config/src/inline/natspec.rs

```diff
@@ -0,0 +1,207 @@
+    pub fn config_lines_with_prefix<S: Into<String>>(&self, prefix: S) -> Vec<String> {
+        let prefix: String = prefix.into();
+        self.config_lines().into_iter().filter(|l| l.starts_with(&prefix)).collect()
+    }
+
+    /// Returns a list of all the configuration lines available in the natspec
+    pub fn config_lines(&self) -> Vec<String> {
+        self.docs
+            .split('\n')
+            .map(remove_whitespaces)
+            .filter(|line| line.contains(INLINE_CONFIG_PREFIX))
+            .collect::<Vec<String>>()
+    }
+}
+
+/// Given a list of nodes, find a "ContractDefinition" node that matches
+/// the provided contract_id.
+fn contract_root_node<'a>(nodes: &'a [Node], contract_id: &'a str) -> Option<&'a Node> {
+    for n in nodes.iter() {
+        if let NodeType::ContractDefinition = n.node_type {
+            let contract_data = &n.other;
+            if let Value::String(contract_name) = contract_data.get("name")? {
+                if contract_id.ends_with(contract_name) {
+                    return Some(n)
+                }
+            }
+        }
+    }
+    None
+}
+
+/// Implements a DFS over a compiler output node and its children.
+/// If a natspec is found it is added to `natspecs`
+fn apply(natspecs: &mut Vec<NatSpec>, contract: &str, node: &Node) {
+    for n in node.nodes.iter() {
+        if let Some((function, docs, line)) = get_fn_data(n) {
+            natspecs.push(NatSpec { contract: contract.into(), function, line, docs })
+        }
+        apply(natspecs, contract, n);
+    }
+}
+
+fn get_fn_data(node: &Node) -> Option<(String, String, String)> {
+    if let NodeType::FunctionDefinition = node.node_type {
+        let fn_data = &node.other;
+        let fn_name: String = get_fn_name(fn_data)?;
+        let (fn_docs, docs_src_line): (String, String) = get_fn_docs(fn_data)?;
+        return Some((fn_name, fn_docs, docs_src_line))
+    }
+
+    None
+}
+
+fn get_fn_name(fn_data: &BTreeMap<String, Value>) -> Option<String> {
+    match fn_data.get("name")? {
+        Value::String(fn_name) => Some(fn_name.into()),
+        _ => None,
+    }
+}
+
+/// Inspects Solc compiler output for documentation comments:
+/// - `Some((String, String))` in case the function has natspec comments. First item is a textual
+///   natspec representation, the second item is the natspec src line, in the form "raw:col:length".
+/// - `None` in case the function has not natspec comments.
+fn get_fn_docs(fn_data: &BTreeMap<String, Value>) -> Option<(String, String)> {
+    if let Value::Object(fn_docs) = fn_data.get("documentation")? {
+        if let Value::String(comment) = fn_docs.get("text")? {
+            if comment.contains(INLINE_CONFIG_PREFIX) {
+                let src_line = fn_docs
+                    .get("src")
+                    .map(Value::to_string)
+                    .unwrap_or(String::from("<no-src-line-available>"));
```

Comment:
Hmm can we add a test were we see this happen? If this would be too cumbersome to reproduce (e.g this would require some weird error happening with the compiler output) we can just add a comment how this is handled.

Reply:
I added a new unit test